tests/ArgentPonyWarcraftClient.Integration.Tests/TestUtilities/ClientCredentialsSource.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+
+namespace ArgentPonyWarcraftClient.Integration.Tests.TestUtilities
+{
+    /// <summary>
+    /// A service that retrieves the cliend ID and secret from environment variables or a file.
+    /// </summary>
+    internal static class ClientCredentialsSource
+    {
+        public static ClientCredentials GetCredentials()
+        {
+            ClientCredentials credentials = GetCredentialsFromEnvironment() ?? GetHardCodedCredentialsForLocal();
```

Comment:
When the `ClientFactory` creates a `WarcraftClient` for these integration tests, it gets the credentials from here.  This will check for environment variables first (which it will find on the CircleCI build server).  Failing that, it will try to load hard-coded values below.  Those aren't supplied here (resulting in a `NoCredentialsException` being thrown by default), but a developer could easily add their own values for local execution of the integration tests.  I thought about loading them from a **.gitignore**d file or leveraging user secrets, but I just wanted to get this change out the door.  That might be cleaner for a follow-up change.